clamav.py

```diff
@@ -12,69 +13,120 @@
+from common import CLAMSCAN_PATH
+from common import FRESHCLAM_PATH
+from common import create_dir
+
+
+RE_SEARCH_DIR = r"SEARCH_DIR\(\"=([A-z0-9\/\-_]*)\"\)"
 
 
 def current_library_search_path():
-    ld_verbose = check_output(["ld", "--verbose"])
-    rd_ld = re.compile("SEARCH_DIR\(\"([A-z0-9/-]*)\"\)")
+    ld_verbose = subprocess.check_output(["ld", "--verbose"]).decode("utf-8")
+    rd_ld = re.compile(RE_SEARCH_DIR)
     return rd_ld.findall(ld_verbose)
 
 
-def update_defs_from_s3(bucket, prefix):
+def update_defs_from_s3(s3_client, bucket, prefix):
     create_dir(AV_DEFINITION_PATH)
-    for filename in AV_DEFINITION_FILENAMES:
-        s3_path = os.path.join(AV_DEFINITION_S3_PREFIX, filename)
-        local_path = os.path.join(AV_DEFINITION_PATH, filename)
-        s3_md5 = md5_from_s3_tags(bucket, s3_path)
-        if os.path.exists(local_path) and md5_from_file(local_path) == s3_md5:
-            print("Not downloading %s because local md5 matches s3." % filename)
-            continue
-        if s3_md5:
-            print("Downloading definition file %s from s3://%s" % (filename, os.path.join(bucket, prefix)))
-            s3.Bucket(bucket).download_file(s3_path, local_path)
-
-
-def upload_defs_to_s3(bucket, prefix, local_path):
-    for filename in AV_DEFINITION_FILENAMES:
-        local_file_path = os.path.join(local_path, filename)
-        if os.path.exists(local_file_path):
-            local_file_md5 = md5_from_file(local_file_path)
-            if local_file_md5 != md5_from_s3_tags(bucket, os.path.join(prefix, filename)):
-                print("Uploading %s to s3://%s" % (local_file_path, os.path.join(bucket, prefix, filename)))
-                s3_object = s3.Object(bucket, os.path.join(prefix, filename))
-                s3_object.upload_file(os.path.join(local_path, filename))
-                s3_client.put_object_tagging(
-                    Bucket=s3_object.bucket_name,
-                    Key=s3_object.key,
-                    Tagging={"TagSet": [{"Key": "md5", "Value": local_file_md5}]}
-                )
+    to_download = {}
+    for file_prefix in AV_DEFINITION_FILE_PREFIXES:
+        s3_best_time = None
+        for file_suffix in AV_DEFINITION_FILE_SUFFIXES:
+            filename = file_prefix + "." + file_suffix
+            s3_path = os.path.join(AV_DEFINITION_S3_PREFIX, filename)
+            local_path = os.path.join(AV_DEFINITION_PATH, filename)
+            s3_md5 = md5_from_s3_tags(s3_client, bucket, s3_path)
+            # sharpr update - This check requires access to the sh-us-control antivirus bucket that we don't have. And it only does anything if the bucket is versioned, which ours isn't.
+            # s3_time = time_from_s3(s3_client, bucket, s3_path)
+
+            # if s3_best_time is not None and s3_time < s3_best_time:
+            #     print("Not downloading older file in series: %s" % filename)
+            #     continue
+            # else:
+            #     s3_best_time = s3_time
```

Comment:
This is my code. This only applies to versioned buckets, and our virus definition bucket is not versioned.